File: backend-python/app/modules/system/controller/common_controller.py
from datetime import UTC, datetime

from fastapi import APIRouter
from fastapi.responses import RedirectResponse, Response

from app.common.api_response import ApiResponse, success

router = APIRouter(tags=["common"])


@router.get("/", include_in_schema=False)
async def root() -> RedirectResponse:
    return RedirectResponse(url="swagger-ui/index.html", status_code=302)


@router.get("/favicon.ico", include_in_schema=False)
async def favicon() -> Response:
    return Response(status_code=204)


@router.get("/health", response_model=ApiResponse)
async def health() -> ApiResponse:
    return success(
        {
            "status": "UP",
            "service": "drip-admin-backend",
            "timestamp": datetime.now(UTC).isoformat().replace("+00:00", "Z"),
        }
    )

File: backend-rust/src/modules/system/controller/common_controller.rs
use super::ok;
use crate::common::ApiResponse;
use crate::modules::system::vo::health_vo::HealthVo;
use axum::Json;
use axum::http::StatusCode;
use axum::response::Redirect;
use chrono::Utc;

pub async fn root() -> Redirect {
    Redirect::temporary("swagger-ui/index.html")
}

pub async fn favicon() -> StatusCode {
    StatusCode::NO_CONTENT
}

pub async fn health() -> Json<ApiResponse<HealthVo>> {
    ok(HealthVo {
        status: "UP".to_string(),
        service: "drip-admin-backend".to_string(),
        timestamp: Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
    })
}
